libs/top_level_metrics.py

```diff
@@ -100,14 +102,35 @@ def calculate_metrics_for_timeseries(
 
     metric_summary = None
     if not metrics.empty:
-        metric_summary = calculate_latest_metrics(metrics, icu_metric_details)
+        metric_summary = calculate_latest_metrics(
+            metrics, icu_metric_details, test_positivity_method
+        )
 
     return metrics, metric_summary
 
 
-def calculate_or_copy_test_positivity(data):
+def _lookup_test_positivity_method(
+    positive_tests_provenance: Optional[str], negative_tests_provenance: Optional[str], log
+) -> TestPositivityRatioMethod:
+    method = None
+    if positive_tests_provenance and positive_tests_provenance == negative_tests_provenance:
```

Comment:
Can you add a comment about when this wouldn't ever be true? after reading into it a bit more I can guess, but some context would be helpful

Reply:
I probably made this function work when provenance is None before I had loading the provenance done correctly, then left it in place because it seemed better to log a warning than raise an exception. Would you prefer an exception?